### PR TITLE
Make helpers thread-safe

### LINKED ISSUE #112: ⚠️ Using both asset pipeline and webpacker is not thread safe
Using both `inline_svg_tag` (asset pipeline) and `inline_svg_pack_tag` (webpacker) in the same application can cause big problems 😬

The helpers were sporadically unable to find the specified SVGs. Sometimes they would work fine, other times they would render the placeholder (`<svg><!-- SVG file not found: logo.svg --></svg>`).

After much investigation, we determined that [the changes that added `inline_svg_tag` and `inline_svg_pack_tag`](https://github.com/jamesmartin/inline_svg/pull/103) (made by me 🤭) **are not thread safe** — specifically, changing `InlineSvg.configuration.asset_finder` within [`with_asset_finder`](https://github.com/jamesmartin/inline_svg/pull/103/files#diff-76810304c6563797234460d5732ea6b3R71-R79) can create a race condition.

This was a severe problem for us, so I [hotfixed it](https://github.com/kylefox/inline_svg/commit/21a9e072f6ea88b044172cfe95dd90ee9065b47a) by using a thread-safe variable (`Thread.current[:inline_svg_asset_finder]`) to set/retrieve the asset finder instead of changing/using `InlineSvg.configuration.asset_finder` directly. It's admittedly a gross hack, but seems to have fixed our issue.

Before I submit this as a PR, I'm wondering if anyone knows of a better fix — cc @jamesmartin?

### PR DESCRIPTION
**Pull request by kylefox**
Fixes #112.

tl;dr — using both `inline_svg_tag` and `inline_svg_pack_tag` is not thread-safe because multiple threads could be modifying `InlineSvg.configuration.asset_finder` (a global variable) simultaneously. This pull request uses thread-local variables to swap the current asset finder, rather than change the global variable directly.

Threads kinda make my head spin, so gotchas & feedback are appreciated 🙏

### CODE REVIEW

**Comment by jamesmartin** [comment]
Tested out locally and can't see any regressions.

**Comment by jamesmartin** [comment]
Thanks again for opening this, @kylefox. ✨

**Comment by jamesmartin** [comment]
Released in [v1.7.0](https://github.com/jamesmartin/inline_svg/releases/tag/v1.7.0).
